### PR TITLE
Update the main doc to v2.10

### PR DESCRIPTION
**Pull request by chenghuzi**
* Update dependencies and the original doc to v2.10. 
* Update main doc translation, i.e., `*.po` files which are not under any `generated` folders to v2.10

### CODE REVIEW

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 1]
```diff
@@ -54,6 +54,81 @@ msgstr "显示异常处理和堆栈跟踪信息。"
 msgid "DB URL. (e.g. sqlite:///example.db)"
 msgstr "数据库 URL. (比如 ``sqlite:///example.db`` )"
 
+msgid "ask"
+msgstr ""
+
+#: ../../<_Ask>:1
+msgid "Create a new trial and suggest parameters."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po]: missing translations?

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 2]
```diff
@@ -54,6 +54,81 @@ msgstr "显示异常处理和堆栈跟踪信息。"
...
+msgstr "JSON 格式的搜索空间. 键为名, 值为 :func:`~optuna.distributions.distribution_to_json` 的输出."
+
+#: ../../<_Ask>:1 ../../<_BestTrial>:1 ../../<_BestTrials>:1 ../../<_Studies>:1
+#: ../../<_Trials>:1
+msgid "Output format."
+msgstr "输出格式."
+
+#: ../../<_Ask>:1
+msgid "Flatten nested columns such as params."
+msgstr "将多层的列展, 例如参数, 进行展平."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po]: `将多层的嵌套列, 例如参数, 进行展平. `

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 3]
```diff
@@ -54,6 +54,81 @@ msgstr "显示异常处理和堆栈跟踪信息。"
...
+
+#: ../../<_Ask>:1 ../../<_BestTrial>:1 ../../<_BestTrials>:1
+#: ../../<_CreateStudy>:1 ../../<_Dashboard>:1 ../../<_DeleteStudy>:1
+#: ../../<_StorageUpgrade>:1 ../../<_Studies>:1 ../../<_StudyOptimize>:1
+#: ../../<_StudySetUserAttribute>:1 ../../<_Tell>:1 ../../<_Trials>:1
+msgid "This command is provided by the optuna plugin."
+msgstr "该命令由 optuna 插件提供。"
+
+msgid "best-trial"
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po:107]: `最佳trial`

match the blow

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 4]
```diff
@@ -54,6 +54,81 @@ msgstr "显示异常处理和堆栈跟踪信息。"
...
+msgid "Show the best trial."
+msgstr "展示最佳 trial."
+
+#: ../../<_BestTrial>:1
+msgid "The name of the study to get the best trial."
+msgstr "用于获取最佳 trial 的 study 名."
+
+#: ../../<_BestTrial>:1 ../../<_BestTrials>:1 ../../<_Trials>:1
+msgid "Flatten nested columns such as params and user_attrs."
+msgstr "将多层的列, 例如参数和 user_attrs 展平."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po]: 将多层的嵌套列, 例如参数和 user_attrs 展平.

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 5]
```diff
@@ -54,6 +54,81 @@ msgstr "显示异常处理和堆栈跟踪信息。"
...
+
+msgid "best-trials"
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po:122]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 6]
```diff
@@ -54,6 +54,81 @@ msgstr "显示异常处理和堆栈跟踪信息。"
...
+
+#: ../../<_BestTrials>:1
+msgid "Show a list of trials located at the Pareto front."
+msgstr "展示位于帕累托前沿的 trial 列表."
+
+#: ../../<_BestTrials>:1
+msgid "The name of the study to get the best trials (trials at the Pareto front)."
+msgstr "用于提取最佳 trial (那些位于帕累托前沿上的 trial) 的 study 名 "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po]: `最佳 trial (那些位于帕累托前沿上的 trial) 的 study 名`

I think this is more nature

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 7]
```diff
@@ -149,48 +224,8 @@ msgid "Show a list of studies."
...
-msgid "whether to disable indenting the JSON"
-msgstr "设定是否在 json输出格式中禁用缩进。"
-
-#: ../../<_Studies>:1
-msgid ""
-"Maximum display width, <1 to disable. You can also use the "
-"CLIFF_MAX_TERM_WIDTH environment variable, but the parameter takes "
-"precedence."
+msgid "Flatten nested columns such as directions."
 msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po]: missing translation

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 8]
```diff
@@ -248,6 +283,81 @@ msgstr "设定用户属性中的键 (key)."
 msgid "Value to be set."
 msgstr "设定用户属性中的值 (value)."
 
+msgid "tell"
+msgstr ""
+
+#: ../../<_Tell>:1
+msgid "Finish a trial created with the ask command."
+msgstr "使用 ask 命令来结束一个 trial."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po]: 结束一个 ask命令创建的trial.

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po [hunk 9]
```diff
@@ -248,6 +283,81 @@ msgstr "设定用户属性中的键 (key)."
...
+
+#: ../../<_Tell>:1
+msgid "Trial number."
+msgstr "Trial 数."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/cli.po]: Trial号

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 2.8.0\n"
+
+#: ../../source/reference/generated/optuna.copy_study.rst:2
+msgid "optuna.copy\\_study"
+msgstr ""
+
+#: of optuna.study.study.copy_study:1
+msgid "Copy study from one storage to another."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po]: missing

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+#: of optuna.study.study.copy_study:3
+msgid ""
+"The direction(s) of the objective(s) in the study, trials, user "
+"attributes and system attributes are copied."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po]: missing

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+#: of optuna.study.study.copy_study:7
+msgid "示例"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po:35]: shouldn't we have English here?

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+msgstr ""
+
+#: of optuna.study.study.copy_study
+msgid "参数"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po:39]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 2.8.0\n"
+
+#: ../../source/reference/generated/optuna.copy_study.rst:2
+msgid "optuna.copy\\_study"
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po:22]: The translation and original test in this file are strange

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.pruners.PatientPruner.po [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+#: ../../source/reference/generated/optuna.pruners.PatientPruner.rst:2
+msgid "optuna.pruners.PatientPruner"
+msgstr ""
+
+#: of optuna.pruners._patient.PatientPruner:1
+msgid "Pruner which wraps another pruner with tolerance."
+msgstr ""
+
+#: of optuna.pruners._patient.PatientPruner:4
+msgid "示例"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.pruners.PatientPruner.po:29]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.storages.RetryFailedTrialCallback.po [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+"conditions, such as being preempted by other processes."
+msgstr ""
+
+#: of optuna._callbacks.RetryFailedTrialCallback:9
+msgid "Usage:"
+msgstr ""
+
+#: of optuna._callbacks.RetryFailedTrialCallback
+#: optuna._callbacks.RetryFailedTrialCallback.retried_trial_number
+msgid "参数"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.storages.RetryFailedTrialCallback.po:47]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.storages.fail_stale_trials.po [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+"will be failed, that is, those states will be changed to "
+":obj:`~optuna.trial.TrialState.FAIL`."
+msgstr ""
+
+#: of optuna.storages._heartbeat.fail_stale_trials:8
+msgid "See :class:`~optuna.storages.RDBStorage`."
+msgstr ""
+
+#: of optuna.storages._heartbeat.fail_stale_trials
+msgid "参数"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.storages.fail_stale_trials.po:40]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.study.MaxTrialsCallback.po [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+"number of trials that will be run, you may want to continue running until"
+" you have a certain number of successfullly completed trials or stop the "
+"study when you have a certain number of trials that fail. This "
+":obj:`MaxTrialsCallback` class allows you to set a maximum number of "
+"trials for a particular :class:`~optuna.trial.TrialState` before stopping"
+" the study."
+msgstr ""
+
+#: of optuna._callbacks.MaxTrialsCallback:10
+msgid "示例"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.study.MaxTrialsCallback.po:40]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.study.copy_study.po [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+msgstr ""
+
+#: of optuna.study.study.copy_study:3
+msgid ""
+"The direction(s) of the objective(s) in the study, trials, user "
+"attributes and system attributes are copied."
+msgstr ""
+
+#: of optuna.study.study.copy_study:7
+msgid "示例"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.study.copy_study.po:35]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po [hunk 20]
```diff
@@ -680,22 +683,21 @@ msgid ""
...
 "more sampled than higher values."
-msgstr "如果``log`` 是 "
-"true, 则在一开始的时候,  suggestion 值的范围会被分成宽度为 1 "
-"的网格点. 然后 suggestion "
-"值会被转化到对数域, 并在该域上进行采样. 这些均匀采样的值会被重新转换到原来的线性域上, 并且被舍入到最近的网格点中. 这就是确定采样值的过程. 比如, 如果"
-" `low = 2` 且 `high = 8` "
-"的话,  suggestion 值的范围就是 `[2, 3, 4, "
-"5, 6, 7, 8]` 而且较小的值有更大的概率被采样到. "
+msgstr ""
+"如果``log`` 是 true, 则在一开始的时候,  suggestion 值的范围会被分成宽度为 1 的网格点. 然后 suggestion"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po:681]: Isn't this the same as line 667?

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.create_trial.po [hunk 21]
```diff
@@ -40,36 +40,57 @@ msgid ""
...
+msgid "``params``"
+msgstr ""
+
+#: of optuna.trial._frozen.create_trial:39
+msgid "``distributions``"
+msgstr ""
+
+#: of optuna.trial._frozen.create_trial:40
+#, fuzzy
+msgid "``value`` or ``values``"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.create_trial.po:63]: The English is broken?

File: source/locale/zh_CN/LC_MESSAGES/reference/visualization/generated/optuna.visualization.matplotlib.plot_pareto_front.po [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+msgstr ""
+
+#: of optuna.visualization.matplotlib._pareto_front.plot_pareto_front:4
+msgid ""
+"Please refer to :func:`optuna.visualization.plot_pareto_front` for an "
+"example."
+msgstr ""
+
+#: of optuna.visualization.matplotlib._pareto_front.plot_pareto_front:7
+msgid "示例"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/visualization/generated/optuna.visualization.matplotlib.plot_pareto_front.po:35]: ditto

**Review by belldandyxtq — CHANGES_REQUESTED** [review]
I have a few questions:
1. Did you include new translations in this PR, because I see some new translations. I thought this is just a auto-generated update PR.
2. I notice some Chinese in the `msgid`, Is this normal?

File: source/locale/zh_CN/LC_MESSAGES/reference/cli.po (hunk 3 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/cli.po:107]: I guess this is an exact command option so probably we don't need to translate it

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po (hunk 12 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.copy_study.po:35]: This is generated automatically by Sphinx. Try searching some other keywords like "返回类型" and you will find more.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po (hunk 20 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.Trial.po:681]: This is the issue we discussed before: sometimes when you get a `note` in the original doc, you'll have two duplicating `msgid`s for them and the second one is the one that determines the translation.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.create_trial.po (hunk 21 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.trial.create_trial.po:63]: Yes, due to the `#, fuzzy`. But now it's fixed.

**Review by belldandyxtq — APPROVED** [review]

**Pull request merged**